### PR TITLE
feat: upgraded springboot to 3.3.1 and bumped up other dependencies

### PR DESCRIPTION
**Pull request by amardeep2006**
feat: bumped up dependencies for Java Quickstarts , Springboot 2 is out of support so i have changes all the projects to use latest springboot 3 version available on the day of raising PR. I have tested all projects locally and github actions as well.
docs: Updated read me files to use Java 17 as baseline
Signed-off-by: amardeep2006 <amardeep2006@gmail.com>

# Description

_Please explain the changes you've made_
Upgraded Java Quickstarts to latest versions of dependencies.
## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_
https://github.com/dapr/quickstarts/issues/1042
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

**Comment by amardeep2006** [comment]
@paulyuk @msfussell Please have a look at the PR once you get chance. I have tested the build via github action in my repo fork as well.

**Comment by msfussell** [comment]
@amardeep2006 - Thanks for this PR. Will take a look. @salaboy - Please also review this.

**Review by salaboy — APPROVED** [review]
This looks good to me, thanks a lot @amardeep2006 for these changes.

**Comment by salaboy** [comment]
@msfussell looks good to me, I've approved this PR

File: configuration/java/http/README.md [hunk 1]
```diff
@@ -13,10 +13,10 @@ This quickstart includes one service:
 ## Prerequisites
 
 - [Maven 3.x](https://maven.apache.org/install.html)
-- Java JDK 11 (or greater):
-  - [Microsoft JDK 11](https://docs.microsoft.com/en-us/java/openjdk/download#openjdk-11)
-  - [Oracle JDK 11](https://www.oracle.com/technetwork/java/javase/downloads/index.html#JDK11)
-  - [OpenJDK 11](https://jdk.java.net/11/)
+- Java JDK 17 (or greater):
+  - [Microsoft JDK 17](https://learn.microsoft.com/en-us/java/openjdk/download#openjdk-17)
```
Comment (paulyuk) [inline configuration/java/http/README.md:17]: I was surprised Java 21 LTS is not the most used, but looking at State of java reports, those agree Java 17 is correct for right now.

**Review by paulyuk — APPROVED** [review]
This is an generous contribution, thank you for this.

**Pull request merged**